src/plugins/mf_priority.cpp

```diff
@@ -96,24 +101,29 @@ static void rec_update_cb (flux_t *h,
                            const flux_msg_t *msg,
                            void *arg)
 {
-    char *uid, *fshare, *bank, *default_bank;
+    char *uid, *fshare, *bank, *default_bank, *max_jobs;
 
-    if (flux_request_unpack (msg, NULL, "{s:s, s:s, s:s, s:s}",
+    if (flux_request_unpack (msg, NULL, "{s:s, s:s, s:s, s:s, s:s}",
                              "userid", &uid,
                              "bank", &bank,
                              "default_bank", &default_bank,
-                             "fairshare", &fshare) < 0) {
+                             "fairshare", &fshare,
+                             "max_jobs", &max_jobs) < 0) {
         flux_log_error (h, "failed to unpack custom_priority.trigger msg");
         goto error;
     }
 
     if (flux_respond (h, msg, NULL) < 0)
         flux_log_error (h, "flux_respond");
 
-    if (strcmp (bank, default_bank) == 0)
-        users[std::atoi (uid)]["default"] = std::stod (fshare);
+    struct bank_info b;
+
+    b.fairshare = std::atof (fshare);
+    b.max_jobs = std::atoi (max_jobs);
+    b.current_jobs = 0;
```

Comment:
Also, I'm probably confused, but are you sure you want to reset `current_jobs` to zero when a user/bank is updated? It seems like if there are current active jobs for this user/bank the count should be left alone.

Reply:
Oh, no, I think you're absolutely right. Thanks for pointing this out. Every time there is an update, the `current_jobs` count would be reset to 0. I failed to realize that it already gets initialized to 0 when the `bank_info` object is created. So I should leave the `current_jobs` count alone.

Maybe I can add another payload update towards the end of the sharness test to confirm that `current_jobs` doesn't get reset when another update occurs. Maybe between the last two tests... 🤔

Reply:
A good functionality test might be to send an update and make sure the max_jobs limit is applied, then send another update increasing the max_jobs limit, e.g. by 1, and ensure the new limit is applied. This will check that limits can be updated, and also that sending an update doesn't reset the current job count in the plugin.

Reply:
I like that idea. I think the latest update I pushed up addresses that suggestion. A new payload update gets sent to the plugin with a new `max_jobs` limit of 4 (+1 to the user's previous max jobs limit of 3). This update occurs while the user has 3 active jobs running, so when the user submits another job (to increase their active count to 4), it should be submitted and run successfully. I added another job submission attempt as well to make sure that the user can't submit 5 or more active jobs. I think this should test two things:

1) that the user's `max_jobs` count can be increased while the user has active jobs running, and
2) that the user's `current_jobs` count is not reset when new payload data is sent to the plugin.